docs/how-to-guides/data-basics/2-building-a-list-of-pages.md

```diff
@@ -0,0 +1,608 @@
+inside the list of `useSelect` dependencies to make sure `getEntityRecords` is re-run when the `searchTerm` changes.
+
+Finally, here’s how `MyFirstApp` looks like once we wire it all together:
+
+```js
+const { SearchControl } = wp.components;
+function MyFirstApp() {
+	const [searchTerm, setSearchTerm] = wp.element.useState( '' );
+	const pages = wp.data.useSelect( select => {
+		const query = {};
+		if ( searchTerm ) {
+			query.search = searchTerm;
+		}
+		return select( wp.coreData.store ).getEntityRecords( 'postType', 'page', query );
+	}, [searchTerm] );
+
+	return (
+		<div>
+			<SearchControl
+				onChange={ setSearchTerm }
+				value={ searchTerm }
+			/>
+			<PagesList pages={ pages }/>
+		</div>
+	)
+}
+```
+
+<details><summary>Compiled snippet</summary>
+
+```js
+function MyFirstApp() {
+	const [searchTerm, setSearchTerm] = wp.element.useState( '' );
+	const pages = wp.data.useSelect( select => {
+		const query = {};
+		if ( searchTerm ) {
+			query.search = searchTerm;
+		}
+		return select( wp.coreData.store ).getEntityRecords( 'postType', 'page', query );
+	}, [searchTerm] );
+
+	return (
+		wp.element.createElement( 'div', {},
+			wp.element.createElement( wp.components.SearchControl, {
+				onChange: setSearchTerm,
+				value: searchTerm,
+			} ),
+			wp.element.createElement( PagesList, { pages } ),
+		)
+	)
+}
+```
+
+</details>
+
+Voila! We can now filter the results:
+
+![](./media/list-of-pages/filter.jpg)
+
+### Gutenberg data vs working directly with the API
+
+Let’s take a pause for a moment to consider an alternative approach. Imagine we sent the API requests directly:
+
+```js
+function MyFirstApp() {
+	// ...
+	const [pages, setPages] = useState( [] );
+	useEffect( () => {
+		const url = '/wp-json/wp/v2/pages?search=' + searchTerm;
+		wp
+			.apiFetch( { url } )
+			.then( setPages )
+	}, [searchTerm] );
+	// ...
+}
+```
+
+We would need to solve two problems here.
+
+First, out-of-order updates. Searching for „About” would trigger five API requests filtering for A, Ab, Abo, Abou, and
+About. They could finish in a different order than they started. It is possible that _search=A_ would resolve after _
+search=About_ and we’d display the wrong data.
+
+Gutenberg data helps by handling the asynchronous part behind the scenes. `useSelect` remembers the most recent call and
+returns only the data we expect.
+
+Second, every keystroke would trigger an API request. If you typed About, deleted it, and retyped it, it would
+issue 10 requests in total even though we could reuse the data.
+
+Gutenberg data helps by caching the responses to API requests triggered by `getEntityRecords()`  and reuses them on
+subsequent calls. This is especially important when other components rely on the same entity records.
+
+All in all, the built-in utilities are designed to solve the typical problems so that you can focus on your application
```

Comment:
```suggestion
All in all, the utilities built into core-data are designed to solve the typical problems so that you can focus on your application
```